backend/server/webhook.go

```diff
@@ -1671,3 +1862,148 @@ func filterBitbucketBytebaseCommit(list []bitbucket.WebhookCommit) []bitbucket.W
+	trimmed := strings.TrimPrefix(strings.TrimSpace(jdbcURL), "jdbc:")
+	u, err := url.Parse(trimmed)
+	if err != nil {
+		return db.UnknownType, err
+	}
+
+	switch {
+	case strings.HasPrefix(u.Scheme, "mysql"):
+		return db.MySQL, nil
+	case strings.HasPrefix(u.Scheme, "postgresql"):
+		return db.Postgres, nil
+	}
+	return db.UnknownType, nil
+}
+
+// extractMybatisMapperSQL will extract the SQL from mybatis mapper XML.
+func extractMybatisMapperSQL(mapperContent string) (string, error) {
+	mybatisMapperParser := mapperparser.NewParser(mapperContent)
+	mybatisMapperNode, err := mybatisMapperParser.Parse()
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to parse mybatis mapper xml")
+	}
+	var sb strings.Builder
+	err = mybatisMapperNode.RestoreSQL(mybatisMapperParser.GetRestoreContext(), &sb)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to restore mybatis mapper xml")
+	}
+	return sb.String(), nil
+}
+
+// mybatisMapperXMLFileDatum is the metadata of mybatis mapper XML file.
+// It maintains the mybatis mapper XML file path, mapper XML file content, and the corresponding mybatis configuration XML content.
+type mybatisMapperXMLFileDatum struct {
+	// mapperPath is the git ls-tree syntax filepath of the mybatis mapper XML file.
+	mapperPath string
+	// mapperContent is the content of the mybatis mapper XML file.
+	mapperContent string
+	// configPath is the git ls-tree syntax filepath of the mybatis configuration XML file,
+	// it is empty if the mybatis configuration XML file is not found.
+	configPath string
+	// configContent is the content of the mybatis configuration XML file,
+	// it is empty if the mybatis configuration XML file is not found.
+	configContent string
+}
+
+// buildMybatisMapperXMLFileData will build the mybatis mapper XML file data.
+//
+//	ctx: the context.
+//	repo: the repository will be list file tree and get file content from.
+//	commitID: the commitID is the snapshot of the file tree and file content.
+//	mapperFiles: the map of the mybatis mapper XML file path and content.
+func (s *Server) buildMybatisMapperXMLFileData(ctx context.Context, repo *api.Repository, commitID string, mapperFiles map[string]string) ([]*mybatisMapperXMLFileDatum, error) {
+	if len(mapperFiles) == 0 {
+		return []*mybatisMapperXMLFileDatum{}, nil
+	}
+
+	var mybatisMapperXMLFileData []*mybatisMapperXMLFileDatum
+	// isMybatisConfigXMLRegex is the regex to match the mybatis configuration XML file, if it can match the file content,
+	// we regard the file as the mybatis configuration XML file.
+	var isMybatisConfigXMLRegex = regexp.MustCompile(`(?i)http(s)?://mybatis.org/dtd/mybatis-3-config.dtd`)
+	// configPathCache is the cache of the mybatis configuration XML file directory,
+	// the key is the mybatis mapper XML file ls-tree syntax directory, and value is the mybatis configuration XML file ls-tree syntax path.
+	configPathCache := make(map[string]string)
```

Comment:
The reason for adding the cache is to avoid context timeouts as much as possible.